std-bits/table/src/main/java/org/enso/table/data/column/operation/CountNothing.java

```diff
@@ -0,0 +1,64 @@
+package org.enso.table.data.column.operation;
+
+import org.enso.table.data.column.storage.ColumnStorage;
+import org.enso.table.data.column.storage.ColumnStorageWithNothingMap;
+import org.enso.table.data.table.Column;
+import org.graalvm.polyglot.Context;
+
+/** An operation for counting the number of Nothing values in a Column. */
+public class CountNothing {
+  // ** Counts the number of Nothing values in the given column. */
+  public static long apply(Column column) {
+    ColumnStorage storage = column.getStorage();
+    return applyToStorage(storage);
+  }
+
+  // ** Counts the number of Nothing values in the given storage. */
+  public static long applyToStorage(ColumnStorage storage) {
+    if (storage instanceof ColumnStorageWithNothingMap withNothingMap) {
+      return withNothingMap.getIsNothingMap().cardinality();
+    }
+
+    Context context = Context.getCurrent();
+    long count = 0;
+    for (long i = 0; i < storage.getSize(); i++) {
+      if (storage.isNothing(i)) {
+        count += 1;
+      }
+      context.safepoint();
+    }
+    return count;
+  }
+
+  // ** Returns true if all values in the storage are Nothing. */
+  public static boolean anyNothing(ColumnStorage storage) {
+    if (storage instanceof ColumnStorageWithNothingMap withNothingMap) {
+      return withNothingMap.getIsNothingMap().cardinality() > 0;
+    }
+
+    Context context = Context.getCurrent();
+    for (long i = 0; i < storage.getSize(); i++) {
+      if (!storage.isNothing(i)) {
+        return true;
+      }
+      context.safepoint();
+    }
+    return false;
+  }
+
+  // ** Returns true if all values in the storage are Nothing. */
+  public static boolean allNothing(ColumnStorage storage) {
+    if (storage instanceof ColumnStorageWithNothingMap withNothingMap) {
+      return withNothingMap.getIsNothingMap().cardinality() == withNothingMap.getSize();
```

Comment:
```suggestion
      return withNothingMap.getIsNothingMap().nextClearBit(0) != -1;
```
I think that's right and again gives us a short circuit opportunity

Reply:
Cool, I was also thinking if we can do some way to short circuit here and didn't think of this.

Let's just ensure we have test coverage for these `anyNothing` and `allNothing` shorthands so that we ensure these tricks work ok, as with more complexity I'm more worried about off-by-one errors here.

---

Note about `nextClearBit` - the `BitSet` does not really have a notion of 'length', so the `nextClearBit` never returns `-1`, it just returns the first index which does not have a set bit.

So this should probably be something like:
```suggestion
      return withNothingMap.getIsNothingMap().nextClearBit() >= withNothingMap.getSize();
```